src/scheduler/fifo.c

```diff
@@ -2322,3 +2369,307 @@ next_job(status *policy, server_info *sinfo, int flag)
 	return rjob;
 }
 
+/**
+ * @brief
+ *	Updates the scheduler state to the server.
+ *
+ * @param[in] state - status to be updated.
+ *
+ * @par Side Effects:
+ *	None
+ *
+ *
+ */
+void
+update_svr_sched_state(char *state)
+{
+
+	struct	attropl	*attribs, *patt;
+	if (connector < 0)
+		return;
+
+	attribs = (struct  attropl *)calloc(1, sizeof(struct attropl));
+	if (attribs == NULL) {
+		sprintf(log_buffer, "can't update scheduler attribs, calloc failed");
+		log_err(-1, __func__, log_buffer);
+		return;
+	}
+	patt = attribs;
+
+	/* Scheduler State*/
+	patt->name = ATTR_sched_state;
+	patt->value = state;
+	patt->next = NULL;
+
+	pbs_manager(connector,
+			MGR_CMD_SET,
+			MGR_OBJ_SCHED,
+			sc_name,
+			attribs,
+			NULL);
+	free(attribs);
+}
+
+/**
+ * @brief
+ *	Helper function used to copy the attribute values from batch_status to the corresponding
+ *	scheduler global variables which hold its priv_dir, log_dir and partitions
+ *
+ * @param[in] status - populated batch_status after stating this scheduler from server
+ *
+ *
+ * @par Side Effects:
+ *	None
+ *
+ *
+ */
+static void
+sched_settings_frm_svr(struct batch_status *status)
+{
+	struct attrl *attr;
+	char *tmp_priv_dir = NULL;
+	char *tmp_log_dir = NULL;
+	char *tmp_partitions = NULL;
+	struct	attropl	*attribs, *patt;
+	struct batch_status *ss = NULL;
+	extern char *partitions;
+
+	attr = status->attribs;
+	/*
+	 * resetting the following before fetching from batch_status.
+	 */
+	while (attr != NULL) {
+		if (attr->name != NULL && attr->value != NULL) {
+			if (!strcmp(attr->name, ATTR_sched_priv)) {
+				COPY_ATTR_VALUE(tmp_priv_dir, attr->value);
+			} else if (!strcmp(attr->name, ATTR_sched_log)) {
+				COPY_ATTR_VALUE(tmp_log_dir, attr->value);
+			} else if (!strcmp(attr->name, ATTR_partition)) {
+				COPY_ATTR_VALUE(tmp_partitions, attr->value);
+			}
+		}
+		attr = attr->next;
+	}
+
+	if (!dflt_sched) {
+		int err;
+		int priv_dir_update_fail = 0;
+		if ((log_dir != NULL) && strcmp(log_dir, tmp_log_dir) != 0) {
+			(void)snprintf(path_log,  MAXPATHLEN, tmp_log_dir);
+			log_close(1);
+			if (log_open(logfile, path_log) == -1) {
+				struct attropl *patt;
+				/* update the sched_log attribute with its previous value only */
```

Comment:
If we don't update the server back then there is a mismatch of qmgr -c "p sched" before and after the configuration update in case if the new configuration is wrong.
Example: Admin wants to change sched_log directory to /tmp/sched_log. Lets us this directory does not exist, then scheduler switches back to using old directory, but the problem is who will update server back that scheduler is still using the old directory.
Before change:
qmgr -c "p sched" |grep sched_logs|grep sc1
set sched sc1 sched_log = /var/spool/pbs/sched_logs_sc1

After change:
qmgr -c "p sched" |grep sched_logs|grep sc1
set sched sc1 sched_log = /tmp/sched_logs_sc1

Due to the above reasons I think listing or printing scheduler should always show the accurate information. 
Updating the comment of scheduler is not needed here since scheduler started using its previous log directory in case of a failure. I need to update the design document after review.

Reply:
I disagree that we should set the anything on the server back.  I agree that the server should always have accurate information.   We don't second guess the admin.  If the admin set something, we don't want to set it to anything.  The admin told us what they want.  If they set something in error, then we stop scheduling and set the scheduling state to down.  If I recall correctly, this is what the EDD actually says.

Reply:
I think in this case as well I will go by what admin is saying and report error from scheduler if the directories are not accessible. I also think that you please update the design document first because this may lead to discussions and who knows, maybe you don't have to change code at all.

Reply:
Implemented. If Admin sets something wrong then the scheduler comment is updated with the error message and also as @bhroam suggested, making scheduling to false in this case.
Also updated EDD with this change.